test/sql/constraints/foreignkey/test_foreignkey.test

```diff
@@ -3,8 +3,197 @@
 # group: [foreignkey]
 
```

Comment:
Some tests that still appear to be missing:

* What if we, in the same transaction, insert a tuple in the main table and the referencing table that refers to that tuple? Does that work?
* What if we, in the same transaction, delete a tuple in the referencing table and then delete that tuple in the main table? Does that work?
* What if we perform an update on a column that is NOT part of the foreign key in the referencing table?
* What if we perform an update on a column that is NOT part of the primary key in the main table?
* The TPC-H test as mentioned above

Some new tests, based on the code:
* What if we have a table that has multiple indexes not related to the foreign key relationship (e.g. other columns that are UNIQUE, but not part of the key relationship)
 * What if we insert NULL into the foreign key column? (if the key is UNIQUE instead fo a PRIMARY KEY this is allowed)

Reply:
Okay, I will add it
But some tests is already contained.
For example, the following tests is contained in "test_foreignkey.test".

- What if we perform an update on a column that is NOT part of the foreign key in the referencing table
UPDATE song SET songname='C_song_new' WHERE songartist=3;

- What if we perform an update on a column that is NOT part of the primary key in the main table
UPDATE album SET albumcover='C_cover_new' WHERE artistid=3;

Reply:
Ah you are right, apologies I missed those.

Reply:
Correct me if I'm wrong but I still appear to be missing these tests:

* What if we, in the same transaction, insert a tuple in the main table and the referencing table that refers to that tuple? Does that work?
* What if we, in the same transaction, delete a tuple in the referencing table and then delete that tuple in the main table? Does that work?

Specifically, note that these request the changes to happen *in the same transaction*, i.e. preceded by `BEGIN TRANSACTION`. I see a test that looks like this but does not do this within the same transaction.

* I would also like to see tests involving updates the indexed column of a table with a self-referential foreign key, and then doing a query that involves an index look-up (e.g. `SELECT * FROM tbl WHERE index_key=1` ).